### DISCUSSION TITLE
probabilistic forecasting - concepts and definitions

### DISCUSSION THREAD

**Discussion by fkiraly** [alan-turing-institute/sktime#2758] [Q&A]
Question from @alinaami from discord:

Hi! I am currently writing my thesis with applications of probabilistic forecasting. Are there any scientific papers or further documentation which explains your approach to interval/quantile/variance forecasts in detail? (apart from the pydata berlin notebook (: ) It is not completely clear to me how the intervals are calculated and which role confidence intervals play... are the terms confidence/prediction interval used synonymously in the code? (I am particularly interested in ARIMA/AutoARIMA/AutoETS) Any help is much appreciated, thanks in advance

**Comment by fkiraly** [comment]
regarding the "how computed", for probabilistic forecasts, there are two common cases:
- an estimator already produces them. In this case, it is mostly an interfaced estimator, so the algorithm to create the probabilistic forecasts is usually described in the literature reference for the algorithm, as implemented in some other python package. For instance, the documentation of statsmodels.
- a wrapper which adds the capability for probabilistic forecasts to a estimator, e.g., NaiveVariance or ConformalInterval. In these cases, typically, a paper is referenced, or a formula is given in the docstring.
(generally, if unclear, please raise an issue! e.g., if you think math is missing or wrong, or a reference is missing or wrong)

Regarding the "what they are", i.e., the general conceptual model for what the predictions are, this is currently best explained in the pydata Berlin tutorial (part 2), which we'll try to move to the base documentation at some point where we have time (cough)
I hope it is sufficiently precise mathematically in the notebook. For instance, it is explained there that the intervals are all (supposed to be) predictive intervals (in the frequentist sense), not confidence or credibility intervals on a mean prediction, or a Bayesian parameter posterior.
Again, if it appears to you that the type of interval produced is not a predictive interval but sth else, kindly feel free to raise an issue, because it might be a bug.

Generally, confidence intervals and prediction intervals are not synonymous concepts - the statements/definitions attached to them are in general different. For example, a sensible confidence interval for the predictive mean of a supervised regression model will approach zero width as training sample size approaches infinite, whereas a sensible predictive interval will not (but approach the generative conditional distribution's interval width).

**Reply by TNTran92** [reply]
I have a similar confusion. After looking at the notebook, I still have these questions:

1/ In the `predict_interval` method, for example, if I have `coverage` = 0.8, does it mean the method will generate an upper and lower bound in which 80% of data will lie within?
2/ How is predictive interval implemented in sktime? 
3/ What is the appropriate way to calculate confidence interval?
